### PR TITLE
feat: enable group by on non-ref fields. closes #3822

### LINKED ISSUE #3822: feat: as graphql _groupBy user I can also use non-ontology tables for aggregation
**Is your feature request related to a problem? Please describe.**
In dashboards we want to enable group by on other fields.

**Describe the solution you'd like**
- [ ] expand graphql api to allow group by fields to of all types (remove the ontology filter)
- [ ] check that the SqlQuery indeed handles those fields as group by fields
- [ ] check and fix tests that might not cover these (so: refs, ref_array and primitive types/arrays)

**Describe alternatives you've considered**
We discussed that in long term we would like to merge _agg and _groupBy api

**Additional context**
probably start here  https://github.com/molgenis/molgenis-emx2/blob/4081ec3f12881cb39d323d9cdd9a3216f594b775/backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java#L321

### PR DESCRIPTION
**Pull request by mswertz**
closes #3822

What are the main changes you did:
- enable group by to work on non-ref fields. Required only extension of the graphql api
- efactored graphql api so the field creation can be reused (future: merge the sample and sample_groupBy)

how to test:
- see test

todo:
- [ ] updated docs in case of new feature 
- [ ] added/updated tests (need permission test)

### CODE REVIEW

File: backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java [hunk 1]
```diff
@@ -147,140 +147,143 @@ private GraphQLNamedOutputType createTableObjectType(TableMetadata table) {
       // build the object
       GraphQLObjectType.Builder tableBuilder = GraphQLObjectType.newObject().name(tableObjectType);
       for (Column col : table.getColumnsWithoutHeadings()) {
-        String id = col.getIdentifier();
```
Comment (mswertz) [inline backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java:150]: this was just a refactor

File: backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java [hunk 2]
```diff
@@ -147,140 +147,143 @@ private GraphQLNamedOutputType createTableObjectType(TableMetadata table) {
...
-        }
+        createTableField(col, tableBuilder);
       }
       tableTypes.put(tableObjectType, tableBuilder.build());
     }
     return tableTypes.get(tableObjectType);
   }
 
+  private void createTableField(Column col, GraphQLObjectType.Builder tableBuilder) {
+    String id = col.getIdentifier();
```
Comment (mswertz) [inline backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java:158]: this also

File: backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java [hunk 3]
```diff
@@ -319,11 +322,11 @@ private GraphQLNamedOutputType createTableGroupByType(TableMetadata table) {
 
       for (Column column : table.getColumns()) {
         // for now only 'ref' types. We might want to have truncating actions for the other types.
-        if (column.isReference() && (hasViewPermission(table) || column.isOntology())) {
-          groupByBuilder.field(
-              GraphQLFieldDefinition.newFieldDefinition()
-                  .name(column.getIdentifier())
-                  .type(createTableObjectType(column.getRefTable())));
+        if (column.isReference()
+                && (hasViewPermission(table) && hasViewPermission(column.getRefTable())
```
Comment (mswertz) [inline backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java:326]: this is the real change

File: backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java [hunk 4]
```diff
@@ -324,6 +327,8 @@ private GraphQLNamedOutputType createTableGroupByType(TableMetadata table) {
               GraphQLFieldDefinition.newFieldDefinition()
                   .name(column.getIdentifier())
                   .type(createTableObjectType(column.getRefTable())));
+        } else if (!column.isReference() && hasViewPermission(table)) {
```
Comment (mswertz) [inline backend/molgenis-emx2-graphql/src/main/java/org/molgenis/emx2/graphql/GraphqlTableFieldFactory.java:330]: this is the actual change :-)

**Review by davidruvolo51 — APPROVED** [review]
This is great. I'm able to run queries on non-ontology/ref types.

```js
{
  Pet_groupBy {
    category {
      name
    }
    count
  }
}
```

```js
{
  Pet_groupBy {
    status
    count
  }
}
```

**Pull request merged**